### ISSUE TITLE
after upgrading to 21.9.0, see illegal state exception: "Unable to perform aggregation for sync committee"

### ISSUE THREAD

**Issue by rolfyone** [Consensys-Incorporated/teku#4370]

**Comment by benjaminion** [comment]
Context, from [Discord](https://discord.com/channels/697535391594446898/697539289042649190/887602196454187048)

![image](https://user-images.githubusercontent.com/20796281/133404270-1503a7b1-2cc2-4fbd-b81f-00721f12fc2e.png)

**Comment by ajsutton** [comment]
The key exception message from the full logs:
```
2021-09-15 07:23:16.521+00:00 | TimeTickChannel-0 | INFO  | teku-event-log | Slot Event  *** Slot: 1265216, Block: 7b8502..d481, Epoch: 39538, Finalized checkpoint: 39536, Finalized root: 0629f8..8888, Peers: 39
2021-09-15 07:23:24.041+00:00 | validator-async-1 | ERROR | teku-validator-log | Validator   *** Failed to produce sync_signature  Slot: 1265201
java.lang.IllegalStateException: Chain head is not available or has advanced beyond slot 1265201
        at tech.pegasys.teku.validator.client.duties.synccommittee.SyncCommitteeScheduledDuties.headUnavailableFailure(SyncCommitteeScheduledDuties.java:88) ~[teku-validator-client-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.validator.client.duties.synccommittee.SyncCommitteeScheduledDuties.lambda$performProductionDuty$1(SyncCommitteeScheduledDuties.java:82) ~[teku-validator-client-21.9.1.jar:21.9.1]
        at java.util.Optional.orElseGet(Optional.java:369) ~[?:?]
        at tech.pegasys.teku.validator.client.duties.synccommittee.SyncCommitteeScheduledDuties.performProductionDuty(SyncCommitteeScheduledDuties.java:82) ~[teku-validator-client-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.validator.client.PendingDuties.lambda$onProductionDue$2(PendingDuties.java:71) ~[teku-validator-client-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.validator.client.PendingDuties.lambda$processPendingActions$7(PendingDuties.java:134) ~[teku-validator-client-21.9.1.jar:21.9.1]
        at java.util.ArrayList.forEach(ArrayList.java:1541) ~[?:?]
        at tech.pegasys.teku.validator.client.PendingDuties.lambda$processPendingActions$8(PendingDuties.java:134) ~[teku-validator-client-21.9.1.jar:21.9.1]
        at java.util.Optional.ifPresent(Optional.java:183) ~[?:?]
        at tech.pegasys.teku.validator.client.PendingDuties.processPendingActions(PendingDuties.java:134) ~[teku-validator-client-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$finish$22(SafeFuture.java:306) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:930) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:907) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:147) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:147) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
        at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:883) ~[?:?]
        at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2251) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:529) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:32) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateResult(SafeFuture.java:142) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$exceptionallyCompose$27(SafeFuture.java:367) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:147) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
        at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:883) ~[?:?]
        at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2251) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:529) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:32) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateResult(SafeFuture.java:142) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$exceptionallyCompose$27(SafeFuture.java:367) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:147) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateToAsync$13(SafeFuture.java:269) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:81) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$2(AsyncRunner.java:38) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:73) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:119) ~[teku-infrastructure-async-21.9.1.jar:21.9.1]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
```

Notably it's trying to create a sync signature for a pretty old slot - the one that was current right as the node started up.  But there are no active validators (one not yet deposited and one in the pending queue) so there shouldn't have been any scheduled duties.

**Comment by chrishobcroft** [comment]
FWIW, further context on this.

The specific use case I was testing on `teku-testnet` server (logs above) was that of running 2 Validator keys, where 1 is active and 1 is inactive. This is to verify whether it is safe to:

1. Configure the software to recognise `n+m` validators (`n` and `m` positive integers), where `n` validators are active and have deposited, and `m` are inactive and are yet to be deposited.
2. Run with `n` active (and `m` inactive) for some time
3. Deposit to for the `m` inactive validators, and test their successful activation

**Comment by chrishobcroft** [comment]
@rolfyone please reopen this. I am able to replicate the issue with `v21.9.1+10-gc9d9d08`:

![image](https://user-images.githubusercontent.com/2212651/133586393-f4c04df6-9325-4442-aff9-d0f4b367cf7b.png)

It also now appears to be throwing this error for every slot event.

**Comment by chrishobcroft** [comment]
See attached `teku.log` file for full stack trace: [teku.log](https://github.com/ConsenSys/teku/files/7176677/teku.log)

**Comment by rolfyone** [comment]
I was actually able to reproduce this, admittedly with active validators, but the theory is the same.
Basically if there are none of your validators in the current sync committee, you will get this IllegalStateException appearing. 
On the plus side, while it's unsightly, it's not causing actual data problems or network traffic, but we really shouldn't be showing this error, because basically there's nothing to do and we try to perform tasks without any validators to work with.
I really appreciate the retest, i wasn't able to replicate it yesterday, but thankfully we got another way to get into that scenario today, so i was able to see the problem and prove it has been resolved with the code build from #4380
